File: src/main/java/com/gmail/samueler53/fwhumanstratego/gui/SetArenaGui.java
package com.gmail.samueler53.fwhumanstratego.gui;

import com.github.stefvanschie.inventoryframework.Gui;
import com.github.stefvanschie.inventoryframework.GuiItem;
import com.github.stefvanschie.inventoryframework.pane.OutlinePane;
import com.gmail.samueler53.fwhumanstratego.managers.ArenaManager;
import com.gmail.samueler53.fwhumanstratego.objects.Arena;
import org.bukkit.Bukkit;
import org.bukkit.Color;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.inventory.meta.LeatherArmorMeta;

import java.util.Objects;
import java.util.UUID;

public class SetArenaGui {


    private final Gui setGui = new Gui(3, "Role");
    private UUID u;
    private Arena arena;

    public static SetArenaGui getInstance() {
        return new SetArenaGui();
    }

    public Gui SetGui() {

        setGui.setOnGlobalClick(event -> event.setCancelled(true));

        OutlinePane background = new OutlinePane(0, 0, 9, 3);
        background.addItem(new GuiItem(new ItemStack(Material.BLACK_STAINED_GLASS_PANE)));
        background.setRepeat(true);
        setGui.addPane(background);


        OutlinePane rolePane = new OutlinePane(2, 1, 5, 1);

        ItemStack redTeamStack = redTeamItemStack();
        ItemStack blueTeamStack = blueTeamItemStack();
        ItemStack treasureRedStack = treasureRedItemStack();
        ItemStack treasureBlueStack = treasureBlueItemStack();
        ItemStack lobbyStack = lobbyItemStack();
        ArenaManager instanceArena = ArenaManager.getInstance();


        rolePane.addItem(new GuiItem(redTeamStack, event -> instanceArena.setRedTeamLocation(Objects.requireNonNull(Bukkit.getPlayer(u)).getLocation(), arena, u)));
        rolePane.addItem(new GuiItem(blueTeamStack, event -> instanceArena.setBlueTeamLocation(Objects.requireNonNull(Bukkit.getPlayer(u)).getLocation(), arena, u)));
        rolePane.addItem(new GuiItem(treasureRedStack, event -> instanceArena.setTreasureRedLocation(Objects.requireNonNull(Bukkit.getPlayer(u)).getLocation(), arena, u)));
        rolePane.addItem(new GuiItem(treasureBlueStack, event -> instanceArena.setTreasureBlueLocation(Objects.requireNonNull(Bukkit.getPlayer(u)).getLocation(), arena, u)));
        rolePane.addItem(new GuiItem(lobbyStack, event -> instanceArena.setLobbyLocation(Objects.requireNonNull(Bukkit.getPlayer(u)).getLocation(), arena, u)));
        setGui.addPane(rolePane);

        return setGui;
    }

    public ItemStack getMaterial(Color color) {
        ItemStack leatherChestplate = new ItemStack(Material.LEATHER_CHESTPLATE);
        LeatherArmorMeta meta = (LeatherArmorMeta) leatherChestplate.getItemMeta();
        if (meta == null) return leatherChestplate;
        meta.setColor(color);
        leatherChestplate.setItemMeta(meta);
        return leatherChestplate;
    }

    private ItemStack redTeamItemStack() {
        ItemStack redTeamStack = getMaterial(Color.RED);
        ItemMeta redTeamMeta = redTeamStack.getItemMeta();
        if (redTeamMeta == null) return redTeamStack;
        redTeamMeta.setDisplayName("Spawn del team rosso");
        redTeamStack.setItemMeta(redTeamMeta);
        return redTeamStack;
    }

    private ItemStack blueTeamItemStack() {
        ItemStack blueTeamStack = getMaterial(Color.BLUE);
        ItemMeta blueTeamMeta = blueTeamStack.getItemMeta();
        if (blueTeamMeta == null) return blueTeamStack;
        blueTeamMeta.setDisplayName("Spawn del team blu");
        blueTeamStack.setItemMeta(blueTeamMeta);
        return blueTeamStack;
    }

    private ItemStack treasureRedItemStack() {
        ItemStack treasureRedStack = new ItemStack(Material.RED_WOOL);
        ItemMeta treasureRedMeta = treasureRedStack.getItemMeta();
        if (treasureRedMeta == null) return treasureRedStack;
        treasureRedMeta.setDisplayName("Tesoro rosso");
        treasureRedStack.setItemMeta(treasureRedMeta);
        return treasureRedStack;
    }

    private ItemStack treasureBlueItemStack() {
        ItemStack treasureBlueStack = new ItemStack(Material.BLUE_WOOL);
        ItemMeta treasureBlueMeta = treasureBlueStack.getItemMeta();
        if (treasureBlueMeta == null) return treasureBlueStack;
        treasureBlueMeta.setDisplayName("Tesoro blu");
        treasureBlueStack.setItemMeta(treasureBlueMeta);
        return treasureBlueStack;
    }

    private ItemStack lobbyItemStack() {
        ItemStack lobbyStack = new ItemStack(Material.GLASS_PANE);
        ItemMeta LobbyMeta = lobbyStack.getItemMeta();
        if (LobbyMeta == null) return lobbyStack;
        LobbyMeta.setDisplayName("Lobby");
        lobbyStack.setItemMeta(LobbyMeta);
        return lobbyStack;
    }

    public void setPlayer(UUID u) {
        this.u = u;
    }

    public void setArena(Arena arena) {
        this.arena = arena;
    }

}

File: src/main/java/com/gmail/samueler53/fwhumanstratego/gui/SetArenaGui.kt
package com.gmail.samueler53.fwhumanstratego.gui

import com.github.stefvanschie.inventoryframework.Gui
import com.github.stefvanschie.inventoryframework.GuiItem
import com.github.stefvanschie.inventoryframework.pane.OutlinePane
import com.gmail.samueler53.fwhumanstratego.managers.ArenaManager
import com.gmail.samueler53.fwhumanstratego.objects.Arena
import org.bukkit.Color
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.bukkit.inventory.meta.LeatherArmorMeta

object SetArenaGui {

    fun showToPlayer(player: Player, arena: Arena) {
        with(Gui(3, "Role")) {
            setOnGlobalClick { it.isCancelled = true }

            val background = OutlinePane(0, 0, 9, 3).apply {
                addItem(GuiItem(ItemStack(Material.BLACK_STAINED_GLASS_PANE)))
                setRepeat(true)
            }

            addPane(background)

            addPane(OutlinePane(2, 1, 5, 1).apply {
                addItem(GuiItem(redTeamSpawnItemStack) {
                    ArenaManager.setRedTeamLocation(player.location, arena, player)
                })

                addItem(GuiItem(blueTeamSpawnItemStack) {
                    ArenaManager.setBlueTeamLocation(player.location, arena, player)
                })

                addItem(GuiItem(treasureRedItemStack()) {
                    ArenaManager.setTreasureRedLocation(player.location, arena, player)
                })

                addItem(GuiItem(treasureBlueItemStack()) {
                    ArenaManager.setTreasureBlueLocation(player.location, arena, player)
                })

                addItem(GuiItem(lobbyItemStack()) {
                    ArenaManager.setLobbyLocation(player.location, arena, player)
                })
            })
            show(player)
        }
    }

    private fun getTeamSpawnItemStack(
        color: Color,
        displayName: String
    ) = ItemStack(Material.LEATHER_CHESTPLATE).apply {
        itemMeta = (itemMeta as LeatherArmorMeta).apply {
            setDisplayName(displayName)
            setColor(color)
        }
    }

    private val redTeamSpawnItemStack get() = getTeamSpawnItemStack(Color.RED, "Spawn del team rosso")

    private val blueTeamSpawnItemStack get() = getTeamSpawnItemStack(Color.BLUE, "Spawn del team blu")

    private fun treasureRedItemStack() = ItemStack(Material.RED_WOOL).apply {
        itemMeta = itemMeta?.apply {
            setDisplayName("Tesoro rosso")
        }
    }

    private fun treasureBlueItemStack() = ItemStack(Material.BLUE_WOOL).apply {
        itemMeta = itemMeta?.apply {
            setDisplayName("Tesoro blu")
        }
    }

    private fun lobbyItemStack() = ItemStack(Material.GLASS_PANE).apply {
        itemMeta = itemMeta!!.apply {
            setDisplayName("Lobby")
        }
    }
}